blockchain/chainio.go

```diff
@@ -1221,6 +1221,25 @@ func (b *BlockChain) initChainState() error {
 			return err
 		}
 
+		// As a final consistency check, we'll run through all the
+		// nodes which are ancestors of the current chain tip, and mark
+		// them as valid if they aren't already marked as such.  This
+		// is a safe assumption as all the block before the current tip
+		// are valid by definition.
+		for iterNode := tip; iterNode != nil; iterNode = iterNode.parent {
```

Comment:
Seems expensive to go through the whole chain every time? Not sure if it matters, as it seems like we are reading the whole chain from disk above anyway.

Reply:
It's just in memory, so it should be pretty quick. This happens on start up only, as we're already reading the entire chain into memory.